File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        diffs = {}

        for i, num in enumerate(nums):
            diff = target - num
            if diff in diffs.keys():
                return [diffs[diff], i]
            diffs.update({num : i})

        return []

File: Data Structures & Algorithms/find-duplicate-integer/submission-0.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        int i = 0;
        while (nums[i] != 0) {
            auto next = nums[i];
            nums[i] = 0;
            i = next;
        }
        return i;
    }
};
